tasks/internal_reconcile_report_orphan/internal_reconcile_report_orphan.py

```diff
@@ -0,0 +1,223 @@
+    """
+    return text(
+        """
+SELECT
+    key_path,
+    etag,
+    (EXTRACT(EPOCH FROM date_trunc('milliseconds', last_update) AT TIME ZONE 'UTC') * 1000)::bigint as last_update,
+    size_in_bytes,
+    storage_class
+    FROM reconcile_orphan_report
+    WHERE job_id = :job_id
+    OFFSET :page_index*:page_size
+    LIMIT :page_size+1"""
+    )
+
+
+def create_http_error_dict(
+    error_type: str, http_status_code: int, request_id: str, message: str
+) -> Dict[str, Any]:
+    """
+    Creates a standardized dictionary for error reporting.
+    Args:
+        error_type: The string representation of http_status_code.
+        http_status_code: The integer representation of the http error.
+        request_id: The incoming request's id.
+        message: The message to display to the user and to record for debugging.
+    Returns:
+        A dict with the following keys:
+            'errorType' (str)
+            'httpStatus' (int)
+            'requestId' (str)
+            'message' (str)
+    """
+    # CumulusLogger will error if a string containing '{' or '}' is passed in without escaping.
+    message = message.replace("{", "{{").replace("}", "}}")
+    LOGGER.error(message)
+    return {
+        "errorType": error_type,
+        "httpStatus": http_status_code,
+        "requestId": request_id,
+        "message": message,
+    }
+
+
+def check_env_variable(env_name: str) -> str:
+    """
+    Checks for the lambda environment variable.
+    Args:
+        env_name (str): The environment variable name set in lambda configuration.
+    Raises: KeyError in case the environment variable is not found.
+    """
+    try:
+        env_value = os.environ[env_name]
+        if len(env_value) == 0 or env_value is None:
+            raise KeyError(f"Empty value for {env_name}")
+    except KeyError:
+        LOGGER.error(f"{env_name} environment value not found.")
+        raise
+
+    return env_value
+
+
+def handler(
+    event: Dict[str, Union[str, int]], context: Any
+) -> Union[List[Dict[str, Any]], Dict[str, Any]]:
+    """
+    Entry point for the internal_reconcile_report_orphan Lambda.
+    Args:
+        event: See schemas/input.json for details.
+        context: An object provided by AWS Lambda. Used for context tracking.
+
+    Environment Vars:
+        DB_CONNECT_INFO_SECRET_ARN (string): Secret ARN of the AWS secretsmanager secret for connecting to the database.
+        See shared_db.py's get_configuration for further details.
+
+    Returns:
+        See schemas/output.json
+        Or, if an error occurs, see create_http_error_dict
+            400 if input does not match schemas/input.json. 500 if an error occurs when querying the database.
+    """
+    try:
+        LOGGER.setMetadata(event, context)
+
+        try:
+            with open("schemas/input.json", "r") as raw_schema:
+                schema = json.loads(raw_schema.read())
+
+            validate = fastjsonschema.compile(schema)
```

Comment:
I thought we wanted to validate every call so that all of them match the schemas.

Since all of our lambdas currently have the same structure (validating for each call), I suggest we create a card for updating all our lambdas to keep the styling consistent if this is the what we want moving forward.
thoughts @AndrewDorn ?

Reply:
will update the code to move out of the handler into main and also create a card to update other lambdas in backlog

Reply:
created https://bugs.earthdata.nasa.gov/browse/ORCA-426